### DISCUSSION TITLE
nextPage doesnt work properly

### DISCUSSION THREAD

**Discussion by OBakir90** [TanStack/table#2731] [Q&A]
Version: v7.5.0 

I created button for each page within ul that goes to the page by using gotoPage when clicked any button. previousPage and nextpage functions works fine at the beginning but after I picked a page by button nextPage doesnt work whereas previousPage works perfectly.

Browser: Chrome

**Comment by umarluqman** [comment]
Are you using uncontrolled pagination? if yes then probably it's because of the pageCount option is a falsy value. I have encountered this before. 

This was the fix for me,
```pageCount: Math.ceil(data.length / size)```

**Comment by OBakir90** [comment]
> Are you using uncontrolled pagination? if yes then probably it's because of the pageCount option is a falsy value. I have encountered this before.
> 
> This was the fix for me,
> `pageCount: Math.ceil(data.length / size)`

Thanks for replying...  yes, I'm using uncontrolled pagination. I tried to use like this:

```
const [pageCounts, setPageCounts] = useState()

    setTimeout(() => {
        setPageCounts(Math.ceil(data.length / pageSize))
    }, [1000])
.....
 useTable({
        columns,
        data,
        initialState: { pageIndex: 0, pageSize: 10 },
        pageCount: { pageCounts }
    }
```
but it doesn't change anything. is this what you mean? or should I change it to controlled pagination?

**Comment by umarluqman** [comment]
There is no need for ``pageCounts local state`` if you are using the build-in pagination (uncontrolled).
Just pass directly to the options, react table will handle it for you

```.....
 useTable({
        columns,
        data,
        initialState: { pageIndex: 0, pageSize: 10 },
        pageCount: Math.ceil(data.length / size) 
    }
```
